### PR TITLE
feat: implelemt rate, increase and delta in PromQL

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- implement `rate()` fn
- implement `delta()` fn
- fix `increase` fn

The current rate of compliance is approximately 37% (⬆️4%).

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
Ref to #1042

### CODE REVIEW

File: src/promql/src/functions/extrapolate_rate.rs [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+
+        // construct matrix from input
+        let ts_array = extract_array(&input[0])?;
+        let ts_range = RangeArray::try_new(ts_array.data().clone().into())?;
+        let value_array = extract_array(&input[1])?;
+        let value_range = RangeArray::try_new(value_array.data().clone().into())?;
+        let ts = extract_array(&input[2])?;
+        let ts = ts
+            .as_any()
+            .downcast_ref::<TimestampMillisecondArray>()
```
Comment (killme2008) [inline src/promql/src/functions/extrapolate_rate.rs:90]: So we don't support time index type to be bigint?

**Review by killme2008 — APPROVED** [review]
LGTM. Great work!

File: src/promql/src/functions/extrapolate_rate.rs [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+            let mut factor = Self::extrapolate_factor(
+                timestamps,
+                end_ts,
+                self.range_length,
+                *values.first().unwrap(),
+                result_value,
+            );
+
+            if IS_RATE {
+                factor /= self.range_length as f64 / 1000.0;
```
Comment (yuanbohan) [inline src/promql/src/functions/extrapolate_rate.rs:136]: Is it possible that the `self.range_length` is zero?

File: src/promql/src/functions/extrapolate_rate.rs (hunk 1 above)
Comment (waynexia) [reply src/promql/src/functions/extrapolate_rate.rs:90]: Yes. At this stage all types are hardcoded. Timestamp can only be `TimestampMillisecond` 😥

File: src/promql/src/functions/extrapolate_rate.rs (hunk 2 above)
Comment (waynexia) [reply src/promql/src/functions/extrapolate_rate.rs:136]: This is checked during parsing. But I should also add an assertion here 👍

**Review by yuanbohan — APPROVED** [review]

**Pull request merged**